contrib/internal/httputil/gen/gen.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"html/template"
+	"os"
+)
+
+func main() {
+	interfaces := []string{"Flusher", "Pusher", "CloseNotifier", "Hijacker"}
+	var combos [][][]string
+	for pick := len(interfaces); pick > 0; pick-- {
+		combos = append(combos, combinations(interfaces, pick))
+	}
+
+	template.Must(template.ParseFiles("./gen.gohtml")).Execute(os.Stdout, map[string]interface{}{
```

Comment:
Thanks for being patient again. And I'm sorry for being unclear. I think it would be nicer and easier to handle if both the template text and the Go code would be in the same file. Basically, at the end of the Go code we'd but a `const template=...` with a backtick, like [here](https://github.com/upspin/upspin/blob/master/version/make_version.go) or [here](https://github.com/gbbr/datadog-trace-agent/blob/master/info/make.go).

Just like in those two examples, we could also just keep this in the `httputil` folder using the `// +build !ignore` flag. All in one file: `make_responsewriter.go` (or similar). I found this worked well in the past, avoiding the need for a new package.